cpp/src/arrow/compute/exec/hash_join_node_test.cc

```diff
@@ -1298,6 +1298,15 @@ void TestHashJoinDictionaryHelper(
     }
   }
 
+  // Instead of sending 2 batches of size 0 we should not send any batches
+  // at all to more accurately simulate real world use cases
+  if (l_length == 0) {
+    l_batches.batches.resize(0);
+  }
+  if (r_length == 0) {
+    r_batches.batches.resize(0);
+  }
+
```

Comment:
Since we now handle both cases (no batches and only empty batches) can we test both scenarios?